### ISSUE TITLE
Warning "unnecessary braces" for expr attribute after updating nightly

### ISSUE THREAD

**Issue by Boscop** [rust-lang/rust#71080]
I just updated my nightly from 02-14 to 04-12, and now I get this warning:
```
warning: unnecessary braces around function argument
    |
281 |             #[auto_enum(Iterator)]
    |             ^^^^^^^^^^^^^^^^^^^^^^ help: remove these braces
    |
    = note: `#[warn(unused_braces)]` on by default
```
for this code using [`auto_enums`](https://github.com/taiki-e/auto_enums):
```rust
	pub fn backup_pixels(&self, iter: impl Iterator<Item = Pixel<Rgb888>>, pixels: &mut Vec<(usize, u8)>) {
		pixels.extend(iter.flat_map(|Pixel(Point { x, y }, _color)| {
			#[auto_enum(Iterator)]
			let r = if (0 .. self.width as _).contains(&x) && (0 .. self.height as _).contains(&y) {
				let index = (x + y * self.width as i32) as usize * 3;
				self.buffer[index .. index + 3].iter().enumerate().map(move |(i, &col)| (index + i, col))
			} else {
				std::iter::empty()
			};
			r
		}));
	}
```
Notice that there are no braces.
It didn't produce a warning with the old nightly.

**Comment by taiki-e** [comment]
Hmm... unused braces generated by macros seem to be properly ignored: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=c4c2befbd2e7486399bfbb521723d0a5

**Comment by Boscop** [comment]
@taiki-e I guess this issue can be closed then?

**Comment by taiki-e** [comment]
Yes, this is due to `#[auto_enum]` incorrectly generates a block that contains the span of the brace of the if expression (i.e., bug of auto_enums crate).

**Comment by Boscop** [comment]
Ah ok.
